### PR TITLE
Sanitize response on `notify` callback

### PR DESCRIPTION
**Pull request by MadLittleMods**
Sanitize response on `notify` callback
 - Lowercase all response data
 - Trim whitespace
 - Return `activate` instead of `Activate` or `Activated`

Covers https://github.com/mikaelbr/node-notifier/issues/101

This is a breaking change.

Here is another upstream change from `toaster` that isn't necessary but will make life easier for consuming the API safely if someone forgets to sanitize: https://github.com/mikaelbr/toaster/pull/2

### CODE REVIEW

**Comment by kurisubrooks** [comment]
To fix the fails in the checks, just add a check for an undefined, and throw an error if so
https://travis-ci.org/mikaelbr/node-notifier/jobs/112464501

**Comment by MadLittleMods** [comment]
@kurisubrooks Thanks for the heads-up, I added a check before sanitizing data as returning `undefined` seems like a valid result.

File: lib/utils.js [hunk 1]
```diff
@@ -161,11 +161,20 @@ module.exports.mapToMac = function (options) {
...
-    if (err || !mapper || !data) return;
-
-    var key = mapper(data);
+  return function (err, data) {
+
+    var resultantData = data;
+    // Sanitize the data
+    if(resultantData) {
+      resultantData = resultantData.toLowerCase().trim();
+      if(resultantData.match(/^activate/)) {
```
Comment (mikaelbr) [inline lib/utils.js]: Should this be case insensitive?

Comment (MadLittleMods) [inline lib/utils.js]: Good call, doesn't hurt to add it in. We use `.toLowerCase()` above so it shouldn't ever affect it.

Comment (mikaelbr) [inline lib/utils.js]: Ah, of course. You are right. Don't mind me then.

File: lib/utils.js [hunk 2]
```diff
@@ -161,11 +161,20 @@ module.exports.mapToMac = function (options) {
...
   options = clone(options);
   fn = fn || function (err, data) {};
-  return function (err, data) {
-    fn.apply(emitter, [err, data]);
-    if (err || !mapper || !data) return;
-
-    var key = mapper(data);
+  return function (err, data) {
+
+    var resultantData = data;
```
Comment (mikaelbr) [inline lib/utils.js]: Why is this needed?

Comment (MadLittleMods) [inline lib/utils.js]: Sometimes `data` can be `undefined` and will throw errors if we try to use string methods on it so we just default the result to the `data` and if it is available, do some work on it and return `resultantData` no matter what happened.

Comment (mikaelbr) [inline lib/utils.js]: Sure, it can be `undefined`, but I was thinking of why storing it in a new variable?
This is effectively the same as:

```js
    // Sanitize the data
    if(data) {
      data = data.toLowerCase().trim();
      if(data.match(/^activate/)) {
        data = 'activate';
      }
    }
    fn.apply(emitter, [err, data]);
    if (err || !mapper || !data) return;

    var key = mapper(data);
    if (!key) return;
    emitter.emit(key, emitter, options);
```

Comment (MadLittleMods) [inline lib/utils.js]: It is the same but I was just keeping it a bit more immutable in a sense and not muddying up the data passed in.

**Comment by MadLittleMods** [comment]
Just a friendly ping, any further review?

**Comment by kurisubrooks** [comment]
@MadLittleMods Mikael can take anywhere from an hour to a week in responding to issues and pull requests. Gotta be patient :)

**Comment by mikaelbr** [comment]
:tada:

**Comment by MadLittleMods** [comment]
@kurisubrooks No worries, I have no problem waiting, just looking for any plans whether it be a week out or whatever, or waiting on some other blocking change, etc.

@mikaelbr Thanks for the merge!

**Comment by sunabozu** [comment]
Would be nice to get this in a release version.

**Comment by kurisubrooks** [comment]
Wasn't aware it wasn't released. Sorry!
ping @mikaelbr

**Comment by mikaelbr** [comment]
I think I'll try to include the terminal-upstream changes before releasing. Will try to do it as soon as I can.

**Comment by gucong3000** [comment]
:+1:
